### ISSUE TITLE
WARNING: mpicc wraps an unrecognized compiler command gcc-5

### ISSUE THREAD

**Issue by zbeekman** [ParaToolsInc/taucmdr#53]
I think taucmdr should be smart enough to allow gcc, gcc-# and gcc-#.# This is probably just a simple regexp addition. Many systems install GCC with suffixes to denote the version if GCC has been updated from the system defaults...

**Comment by zbeekman** [comment]
Furthermore, taucmdr states it will attempt to continue, but then a unit test fails:

```
/runtests
...........................E
Stdout:
[TAU] Probing System MPI C compiler '/usr/local/bin/mpicc' to discover wrapped compiler
[TAU]   '/usr/local/bin/mpicc' wraps '/usr/local/bin/gcc-5'
[TAU] *******************************************************************************************************************************************************************************************************
[TAU]
[TAU] WARNING
[TAU]
[TAU] '/usr/local/bin/mpicc' wraps an unrecognized compiler command '/usr/local/bin/gcc-5'. TAU will attempt to continue but may fail later on.
[TAU]
[TAU] *******************************************************************************************************************************************************************************************************
................................
======================================================================
ERROR: test_no_project (cli.commands.target.tests.test_create.CreateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Volumes/Sandbox/taucmdr/packages/tau/cli/commands/target/tests/test_create.py", line 42, in test_no_project
    self.assertRaises(ProjectStorageError, create_cmd.main, argv)
  File "/usr/local/homebrew/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 475, in assertRaises
    callableObj(*args, **kwargs)
  File "./packages/tau/cli/commands/target/create.py", line 220, in main
    compilers = self.parse_compiler_flags(args)
  File "./packages/tau/cli/commands/target/create.py", line 78, in parse_compiler_flags
    family_comps = InstalledCompilerFamily(family_cls(family_arg))
  File "./packages/tau/cf/compiler/installed.py", line 314, in __init__
    installed = InstalledCompiler(absolute_path, info)
  File "./packages/tau/cf/compiler/installed.py", line 160, in __init__
    self.wrapped = self._probe_wrapper()
  File "./packages/tau/cf/compiler/installed.py", line 199, in _probe_wrapper
    wrapped_command = parts[0]
IndexError: list index out of range

Stdout:
[TAU] Probing System MPI C compiler '/usr/local/bin/mpicc' to discover wrapped compiler
[TAU]   '/usr/local/bin/mpicc' wraps '/usr/local/bin/gcc-5'
[TAU] *******************************************************************************************************************************************************************************************************
[TAU]
[TAU] WARNING
[TAU]
[TAU] '/usr/local/bin/mpicc' wraps an unrecognized compiler command '/usr/local/bin/gcc-5'. TAU will attempt to continue but may fail later on.
[TAU]
[TAU] *******************************************************************************************************************************************************************************************************

----------------------------------------------------------------------
Ran 60 tests in 0.067s

FAILED (errors=1)
WARNING: LevelsTest: tests have not been implemented
WARNING: LocalFileTest: tests have not been implemented
WARNING: ProjectTest: tests have not been implemented
WARNING: ErrorTest: tests have not been implemented
WARNING: LoggerTest: tests have not been implemented
```

**Comment by jlinford** [comment]
New bug sighted!

On Thu, Jun 2, 2016 at 11:46 AM, Izaak Beekman <notifications@github.com>
wrote:

> Furthermore, taucmdr states it will attempt to continue, but then a unit
> test fails:
>
> /runtests
> ...........................E
> Stdout:
> [TAU] Probing System MPI C compiler '/usr/local/bin/mpicc' to discover wrapped compiler
> [TAU]   '/usr/local/bin/mpicc' wraps '/usr/local/bin/gcc-5'
> [TAU] *******************************************************************************************************************************************************************************************************
> [TAU]
> [TAU] WARNING
> [TAU]
> [TAU] '/usr/local/bin/mpicc' wraps an unrecognized compiler command '/usr/local/bin/gcc-5'. TAU will attempt to continue but may fail later on.
> [TAU]
> [TAU] *******************************************************************************************************************************************************************************************************
> ................................
> ======================================================================
> ERROR: test_no_project (cli.commands.target.tests.test_create.CreateTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/Volumes/Sandbox/taucmdr/packages/tau/cli/commands/target/tests/test_create.py", line 42, in test_no_project
>     self.assertRaises(ProjectStorageError, create_cmd.main, argv)
>   File "/usr/local/homebrew/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 475, in assertRaises
>     callableObj(*args, **kwargs)
>   File "./packages/tau/cli/commands/target/create.py", line 220, in main
>     compilers = self.parse_compiler_flags(args)
>   File "./packages/tau/cli/commands/target/create.py", line 78, in parse_compiler_flags
>     family_comps = InstalledCompilerFamily(family_cls(family_arg))
>   File "./packages/tau/cf/compiler/installed.py", line 314, in __init__
>     installed = InstalledCompiler(absolute_path, info)
>   File "./packages/tau/cf/compiler/installed.py", line 160, in __init__
>     self.wrapped = self._probe_wrapper()
>   File "./packages/tau/cf/compiler/installed.py", line 199, in _probe_wrapper
>     wrapped_command = parts[0]
> IndexError: list index out of range
>
> Stdout:
> [TAU] Probing System MPI C compiler '/usr/local/bin/mpicc' to discover wrapped compiler
> [TAU]   '/usr/local/bin/mpicc' wraps '/usr/local/bin/gcc-5'
> [TAU] *******************************************************************************************************************************************************************************************************
> [TAU]
> [TAU] WARNING
> [TAU]
> [TAU] '/usr/local/bin/mpicc' wraps an unrecognized compiler command '/usr/local/bin/gcc-5'. TAU will attempt to continue but may fail later on.
> [TAU]
> [TAU] *******************************************************************************************************************************************************************************************************
>
> ----------------------------------------------------------------------
> Ran 60 tests in 0.067s
>
> FAILED (errors=1)
> WARNING: LevelsTest: tests have not been implemented
> WARNING: LocalFileTest: tests have not been implemented
> WARNING: ProjectTest: tests have not been implemented
> WARNING: ErrorTest: tests have not been implemented
> WARNING: LoggerTest: tests have not been implemented
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ParaToolsInc/taucmdr/issues/53#issuecomment-223333570>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AE7_dqB5X_kplTu7DnBHWh_QQ4nGjerpks5qHvrpgaJpZM4IssTS>
> .
>



-- 
http://www.google.com/profiles/john.christian.linford

**Comment by zbeekman** [comment]
If I swap out MPICH and swap in OpenMPI which is compiled to wrap clang, I get a similar error:
```
$ ./runtests
...........................E
Stdout:
[TAU] Probing System MPI C compiler '/usr/local/bin/mpicc' to discover wrapped compiler
[TAU]   '/usr/local/bin/mpicc' wraps '/usr/local/bin/clang'
[TAU] *******************************************************************************************************************************************************************************************************
[TAU]
[TAU] WARNING
[TAU]
[TAU] '/usr/local/bin/mpicc' wraps an unrecognized compiler command '/usr/local/bin/clang'. TAU will attempt to continue but may fail later on.
[TAU]
[TAU] *******************************************************************************************************************************************************************************************************
................................
======================================================================
ERROR: test_no_project (cli.commands.target.tests.test_create.CreateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Volumes/Sandbox/taucmdr/packages/tau/cli/commands/target/tests/test_create.py", line 42, in test_no_project
    self.assertRaises(ProjectStorageError, create_cmd.main, argv)
  File "/usr/local/homebrew/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 475, in assertRaises
    callableObj(*args, **kwargs)
  File "./packages/tau/cli/commands/target/create.py", line 220, in main
    compilers = self.parse_compiler_flags(args)
  File "./packages/tau/cli/commands/target/create.py", line 78, in parse_compiler_flags
    family_comps = InstalledCompilerFamily(family_cls(family_arg))
  File "./packages/tau/cf/compiler/installed.py", line 314, in __init__
    installed = InstalledCompiler(absolute_path, info)
  File "./packages/tau/cf/compiler/installed.py", line 160, in __init__
    self.wrapped = self._probe_wrapper()
  File "./packages/tau/cf/compiler/installed.py", line 199, in _probe_wrapper
    wrapped_command = parts[0]
IndexError: list index out of range
```

**Comment by zbeekman** [comment]
A quick note: Pretty much every OS X user who uses homebrew to install MPICH or OpenMPI will encounter this issue...

**Comment by jlinford** [comment]
Resolved as of commit 415429a275064e8daf2ceb85bc03b29121faeae8.
